File: CS/GantSolution.Module/BusinessObjects/ITask.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace GanttSolution.Module.BusinessObjects {
    public interface ITask {
        ITask Id { get; }
        string Name { get; set; }
        DateTime StartDate { get; set; }
        DateTime EndDate { get; set; }
        TimeSpan Duration { get; set; }
        double Progress { get; set; }
        ITask Parent { get; set; }
        IList<ITask> PredecessorTasks { get; }
    }
}

File: VB/GanttSolution.Module/BusinessObjects/ITask.vb
Namespace GanttSolution.Module.BusinessObjects
    Public Interface ITask
        ReadOnly Property Id() As ITask
        Property Name() As String
        Property StartDate() As Date
        Property EndDate() As Date
        Property Duration() As TimeSpan
        Property Progress() As Double
        Property Parent() As ITask
        ReadOnly Property PredecessorTasks() As IList(Of ITask)
    End Interface
End Namespace